### DISCUSSION TITLE
ripgrep on non-x86_64 arch?

### DISCUSSION THREAD

**Discussion by Grommish** [BurntSushi/ripgrep#2174] [Q&A] [answered]
I am building for `mips64-openwrt-linux-musl` rust toolchain.  I can successfully cross-compile and deploy other `rust-lang` based projects to the `mips64` arch, but `ripgrep` gives me linking errors. 

Is this just something that isn't designed to work on other archs?  I'm not willing to open an issue on this, since it very well could be an issue on my end (or, again, trying to do something not intended).

**Comment by BurntSushi** [comment, marked as answer]
I can't think of any reason why it wouldn't work if other projects work. ripgrep's default build configuration should be pure Rust too. (Enabling PCRE2 changes that.)

But yeah, I'm not sure how to debug it either. You might post on the Rust user forum, or ask for help on the Rust Discord server.

Also, there are non-x86 architectures included in ripgrep's CI release.

**Reply by Grommish** [reply]
Thanks.. I just wanted to make sure..  I'm seeing linking errors like

```
          /home/grommish/openwrt/staging_dir/toolchain-mips64_octeonplus_64_gcc-11.2.0_musl/lib/gcc/mips64-openwrt-linux-musl/11.2.0/../../../../mips64-openwrt-linux-musl/bin/ld: /home/grommish/openwrt/build_dir/target-mips64_octeonplus_64_musl/ripgrep-13.0.0/target/mips64-openwrt-linux-musl/release/deps/libjemalloc_sys-0a62eebc43d24d05.rlib(jemalloc.pic.o): Relocations in generic ELF (EM: 62)
          /home/grommish/openwrt/staging_dir/toolchain-mips64_octeonplus_64_gcc-11.2.0_musl/lib/gcc/mips64-openwrt-linux-musl/11.2.0/../../../../mips64-openwrt-linux-musl/bin/ld: /home/grommish/openwrt/build_dir/target-mips64_octeonplus_64_musl/ripgrep-13.0.0/target/mips64-openwrt-linux-musl/release/deps/libjemalloc_sys-0a62eebc43d24d05.rlib: error adding symbols: file in wrong format
          collect2: error: ld returned 1 exit status
```

Which suggest an issue with the arch format, so I wanted to make sure there wasn't a reason it *shouldn't* work.  Of the various test `cargo` packages I use and even the hybrid build cases that mix Make/CMake/Rust, this seems to be the one that doesn't want to work right.

I appreciate the insight and thank you again for `ripgrep`!

**Reply by BurntSushi** [reply]
Aye. It could be a bug worth filing against Rust itself.
